lesson4/hometask.rb

```diff
@@ -4,20 +4,27 @@ def method_missing(method_name)
     self[method_name] = fetch(method_name).call
   end
 
-  def respond_to_missing?(method_name)
-    self[method_name] = fetch(method_name).call
+  def respond_to_missing?(method_name, include_private = false)
+    if fetch(method_name)
```

Comment:
It's incorrect. If there is no `:#{method_name}` key, it would not return `false`, it would return an exception.
Easy to check:
```
a = LazyHash.new
a[:b] = Proc.new { 12 }
puts a.respond_to?(:b) # should be true
puts a.respond_to?(:c) # should be false
```
But last line generates exception: 
>in `fetch': key not found: :c (KeyError)

So, you need to handle this. Simple hint: if you have a method which expects to return `true/false` value and use a condition to provide such behaviour, use something that returns a `true/false` as a body of the condition too. `#fetch` is definitely not right choice. Try `#has_role?`. In that case, you won't need even such condition.